### PR TITLE
Influx package

### PR DESCRIPTION
**Pull request by rvolosatovs**
Provides an auto-scaling point writer, which writes points in batches and simple writer, which just writes points one-by-one
see TheThingsIndustries/noc#223 for usage example

### CODE REVIEW

**Comment by htdvisser** [comment]
```
❯ golint influx
influx/writer.go:15:7: exported const DefaultScalingInterval should have comment or be unexported
influx/writer.go:41:1: exported function NewSinglePointWriter should have comment or be unexported
influx/writer.go:110:1: exported function NewBatchingWriter should have comment or be unexported
```

File: influx/writer.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+	w.mutex.RLock()
+	ch, ok := w.pointChans[bpConf]
+	w.mutex.RUnlock()
+	if !ok {
+		w.mutex.Lock()
+		ch, ok = w.pointChans[bpConf]
+		if !ok {
+			ch = make(chan *batchPoint)
+		}
+		w.pointChans[bpConf] = ch
```
Comment (htdvisser) [inline influx/writer.go]: should be inside the `if` on L126

Comment (htdvisser) [inline influx/writer.go]: should be inside the `if` on L126

File: influx/writer.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+		w.mutex.Unlock()
+		go writeInBatches(w.log, w.writer, bpConf, w.scalingInterval, ch)
```
Comment (htdvisser) [inline influx/writer.go]: should be inside the `if` on L126

File: influx/writer.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+			for _, p := range points {
+				bp.AddPoint(p.Point)
+			}
+
+			log.WithField("num", len(points)).Debug("Writing batch of points to Influx")
+			err := w.Write(bp)
+			for _, p := range points {
+				go p.pushError(err)
+			}
+			points = make([]*batchPoint, 0, len(points))
```
Comment (htdvisser) [inline influx/writer.go]:
```go
points = points[:0]
```

Comment (rvolosatovs) [inline influx/writer.go]: nice one!

**Pull request merged**